mbed_host_tests/host_tests_conn_proxy/conn_primitive_remote.py

```diff
@@ -144,6 +154,7 @@ def finish(self):
                     self.selected_resource.closeConnection()
                 if self.selected_resource.is_allocated:
                     self.selected_resource.release()
+                    self.selected_resource = None
```

Comment:
Do you mean user will try again? Can we not make it part of this method. Like line https://github.com/ARMmbed/htrun/pull/152/files#diff-ab29c0280949087d36c3c6cca0af4441R151 can be 
```while self.selected_resource:```